MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java

```diff
@@ -1393,6 +1409,55 @@ protected Optional<JPopupMenu> createPopupMenu() {
+                        if (!tros.contains(Crew.ENVSPC_FOG)) {
+                            menuItem.setActionCommand(makeCommand(CMD_ACQUIRE_ENVSPEC, Crew.ENVSPC_FOG, String.valueOf(cost)));
+                            menuItem.addActionListener(this);
+                            menuItem.setEnabled(available);
+                            specialistMenu.add(menuItem);
+                        }
+                        if (!tros.contains(Crew.ENVSPC_LIGHT)) {
+                            menuItem = new JMenuItem(String.format(resources.getString("abilityDesc.format"), resources.getString("envspec_light.text"), costDesc));
+                            menuItem.setActionCommand(makeCommand(CMD_ACQUIRE_ENVSPEC, Crew.ENVSPC_LIGHT, String.valueOf(cost)));
+                            menuItem.addActionListener(this);
+                            menuItem.setEnabled(available);
+                            specialistMenu.add(menuItem);
+                        }
+                        if (!tros.contains(Crew.ENVSPC_RAIN)) {
+                            menuItem = new JMenuItem(String.format(resources.getString("abilityDesc.format"), resources.getString("envspec_rain.text"), costDesc));
+                            menuItem.setActionCommand(makeCommand(CMD_ACQUIRE_ENVSPEC, Crew.ENVSPC_RAIN, String.valueOf(cost)));
+                            menuItem.addActionListener(this);
+                            menuItem.setEnabled(available);
+                            specialistMenu.add(menuItem);
+                        }
+                        if (!tros.contains(Crew.ENVSPC_SNOW)) {
+                            menuItem = new JMenuItem(String.format(resources.getString("abilityDesc.format"), resources.getString("envspec_snow.text"), costDesc));
+                            menuItem.setActionCommand(makeCommand(CMD_ACQUIRE_ENVSPEC, Crew.ENVSPC_SNOW, String.valueOf(cost)));
+                            menuItem.addActionListener(this);
+                            menuItem.setEnabled(available);
+                            specialistMenu.add(menuItem);
+                        }
+                        if (!tros.contains(Crew.ENVSPC_WIND)) {
+                            menuItem = new JMenuItem(String.format(resources.getString("abilityDesc.format"), resources.getString("envspec_wind.text"), costDesc));
+                            menuItem.setActionCommand(makeCommand(CMD_ACQUIRE_ENVSPEC, Crew.ENVSPC_WIND, String.valueOf(cost)));
+                            menuItem.addActionListener(this);
+                            menuItem.setEnabled(available);
+                            specialistMenu.add(menuItem);
+                        }
+                        if (specialistMenu.getMenuComponentCount() > 0) {
+                            abMenu.add(specialistMenu);
+                        }
```

Comment:
Please use the deprecated JMenuHelpers instead, it handles some issues with this and will simplify migration.

Reply:
Not sure I completely follow, but change line 1459 with something like `JMenuHelpers.addMenuIfNonEmpty(specialistmenu, abMenu);` ?

Further down, line 1584 has `JMenuHelpers.addMenuIfNonEmpty(menu, abMenu);`

Just haven't seen or worked with that function in the code before.

Reply:
It'll be `JMenuHelpers.addMenuIfNonEmpty(abMenu, specialistMenu );`, in order, as that adds the second to the first menu based on if it is empty and then adds scrolling if required.

Reply:
I get the same issue testing with that code.

Reply:
After a little testing, if you check the options for implants **and** edge the glitch isn't there. The weapon specialist menu displays properly. It displays correctly with edge selected alone, but not with implants. Same glitch if only use spas and implants are selected. Edge and SPAs, or all three and the menu has no issues.

Reply:
Wait, this is all a miscommunication. I meant to do this replacement, as the other version is older (ignore the deprecation, there's a further simplification we haven't done yet).

```suggestion
                        JMenuHelpers.addMenuIfNonEmpty(abMenu, specialistMenu);
```

Reply:
That is the same code from above and it does not fix the issue.

`It'll be JMenuHelpers.addMenuIfNonEmpty(abMenu, specialistMenu);, in order, as that adds the second to the first menu based on if it is empty and then adds scrolling if required.`

Reply:
What issues are you having with it? Or is it other code that's causing problems? That should add the menu provided it is filled, and autoscroll it. It doesn't do anything else.

Reply:
It doesn't fix the issue from the image I provided. The scroll bar is there, it is just weapon specialist gets squished for whatever reason.

Reply:
Oh, that's in a totally different file and unrelated to this request. This just improves the code above using the previous code.